pkg/server/sandbox_run.go

```diff
@@ -553,3 +525,48 @@ func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []cni.PortMapping
 	}
 	return portMappings
 }
+
+// untrustedWorkload returns true if the sandbox contains untrusted workload.
+func untrustedWorkload(config *runtime.PodSandboxConfig) bool {
+	return config.GetAnnotations()[annotations.UntrustedWorkload] == "true"
+}
+
+// hostPrivilegedSandbox returns true if the sandbox configuration
+// requires additional host privileges for the sandbox.
+func hostPrivilegedSandbox(config *runtime.PodSandboxConfig) bool {
+	securityContext := config.GetLinux().GetSecurityContext()
+	if securityContext.GetPrivileged() {
+		return true
+	}
+
+	namespaceOptions := securityContext.GetNamespaceOptions()
+	if namespaceOptions.GetNetwork() == runtime.NamespaceMode_NODE ||
+		namespaceOptions.GetPid() == runtime.NamespaceMode_NODE ||
+		namespaceOptions.GetIpc() == runtime.NamespaceMode_NODE {
+		return true
+	}
+
+	return false
+}
+
+// getSandboxRuntime returns the runtime configuration for sandbox.
+// If the sandbox contains untrusted workload, secure runtime will be returned,
+// or else default runtime will be returned.
+func (c *criService) getSandboxRuntime(config *runtime.PodSandboxConfig) (criconfig.Runtime, error) {
+	secureRuntimeRequired := false
+	if untrustedWorkload(config) {
+		// TODO(random-liu): Figure out we should return error or not.
+		if hostPrivilegedSandbox(config) {
+			return criconfig.Runtime{}, errors.New("untrusted workload with host privilege is not allowed")
+		}
+		secureRuntimeRequired = true
+	}
+
+	if secureRuntimeRequired {
+		if c.config.ContainerdConfig.SecureRuntime.Type == "" {
+			return criconfig.Runtime{}, errors.New("no secure runtime is configured")
```

Comment:
Intead of fail because secure runtime was not set. Lets use the default runtime.
```diff
        if secureRuntimeRequired {
                if c.config.ContainerdConfig.SecureRuntime.Type == "" {
-                       return criconfig.Runtime{}, errors.New("no secure runtime is configured")
+                       logrus.Debugf("secure runtime not defined, using default")
+                       return c.config.ContainerdConfig.DefaultRuntime, nil
                }
+               logrus.Debugf("DEBUG: using secure runtime to create Sandbox")
                return c.config.ContainerdConfig.SecureRuntime, nil
        }
        return c.config.ContainerdConfig.DefaultRuntime, nil
 }

```